Core/Resources/Definitions/ZDoom/GldefsDefinition.cs

```diff
@@ -0,0 +1,185 @@
+        {
+            string name = entry.Path.Name;
+            BrightMaps.Auto[name] = new BrightmapDefinition() { TargetTexture = name, BrightmapName = name };
+        }
+    }
+
+    private void ParseEntry(Entry entry)
+    {
+        m_includeStack.Push(entry.Path.FullPath);
+        try
+        {
+
+            string data = entry.ReadDataAsString();
+            SimpleParser parser = new();
+            parser.SetSpecialChars(['{', '}']); // remove most special chars, particularly [ ] since they may be part of a sprite name
+            parser.Parse(data);
+
+            while (!parser.IsDone())
+            {
+                string defType = parser.ConsumeString();
+                if (defType.EqualsIgnoreCase("#include"))
+                    ParseInclude(entry, parser);
+                else if (defType.EqualsIgnoreCase("brightmap"))
+                    ParseBrightmapBlock(entry, parser);
+                else if (defType.EqualsIgnoreCase("skybox"))
+                    ParseSkyboxBlock(entry, parser);
+                else if (!parser.IsDone())
+                    parser.ConsumeLine();
+            }
+        }
+        finally
+        {
+            m_includeStack.Pop();
+        }
+    }
+
+    private void ParseInclude(Entry entry, SimpleParser parser)
+    {
+        string path = parser.ConsumeString();
+        Entry? includeEntry = entry.Parent.Entries.FirstOrDefault(x => x.Path.FullPath.EqualsIgnoreCase(path));
+        if (includeEntry != null)
+        {
+            if (m_includeStack.Contains(includeEntry.Path.FullPath))
+                throw new Exception($"GLDEFS in {entry.Parent.FullPath} contains an infinite loop ({entry.Path.FullPath} -> {includeEntry.Path.FullPath})");
+            ParseEntry(includeEntry);
+        }
+    }
+
+    private void ParseBrightmapBlock(Entry entry, SimpleParser parser)
+    {
+        string textureType = parser.ConsumeString().ToLowerInvariant();
+        string texture = parser.ConsumeString();
+        parser.ConsumeString("{");
+        BrightmapDefinition def = new() { TargetTexture = texture };
+        while (!parser.IsDone())
+        {
+            string token = parser.ConsumeString().ToLowerInvariant();
+            if (token.EqualsIgnoreCase("map"))
+            {
+                string filename = parser.ConsumeString();
+                def.BrightmapName = Path.GetFileNameWithoutExtension(filename);
+            }
+            else if (token.EqualsIgnoreCase("iwad"))
+                def.IwadOnly = true;
+            else if (token.EqualsIgnoreCase("thiswad"))
+                def.SpecificWadMd5 = entry.Parent.MD5;
+            else if (token.EqualsIgnoreCase("disablefullbright"))
+                def.DisableFullbright = true;
+            else if (token == "}")
+            {
+                var destination = textureType switch
+                {
+                    "flat" => BrightMaps.Flats,
```

Comment:
this switch makes it case sensitive so it would have to moved into if statements using EqualsIgnoreCase